### PR TITLE
Extends ChildStateBackend and ChildStateAPI with ReadProofs

### PR DESCRIPTION
**Pull request by mustermeiszer**
The following changes integrate the existing `read_child_proof` from the `ProofProvider` trait into the `ChildStateBackend`. 
To allow the generation of read proofs via RPCs on full nodes, the `ChildStateApi`
is extended with `read_child_proof`.

At the moment, especially useful for parachains who want to retrieve proof of contribution for crowdloan participants.

### CODE REVIEW

File: client/rpc/src/state/mod.rs [hunk 1]
```diff
@@ -466,6 +483,8 @@ impl<Block, Client> ChildStateApi<Block::Hash> for ChildState<Block, Client>
 	) -> FutureResult<Option<u64>> {
 		self.backend.storage_size(block, storage_key, key)
 	}
+
+
```
Comment (bkchr) [inline client/rpc/src/state/mod.rs:487]:
```suggestion
```

File: client/rpc-api/src/child_state/mod.rs [hunk 2]
```diff
@@ -68,4 +69,13 @@ pub trait ChildStateApi<Hash> {
...
 		hash: Option<Hash>
 	) -> FutureResult<Option<u64>>;
+
+	/// Returns proof of storage for child key entries at a specific block's state.
+	#[rpc(name = "state_getChildReadProof")]
+	fn read_child_proof(
+		&self,
+		child_storage_key: PrefixedStorageKey,
+		keys: Vec<StorageKey>,
+		hash: Option<Hash>
```
Comment (bkchr) [inline client/rpc-api/src/child_state/mod.rs:79]:
```suggestion
		hash: Option<Hash>,
```

**Review by bkchr — APPROVED** [review]

File: client/rpc-api/src/child_state/mod.rs (hunk 2 above)
Comment (mustermeiszer) [reply client/rpc-api/src/child_state/mod.rs]: Done

File: client/rpc/src/state/mod.rs (hunk 1 above)
Comment (mustermeiszer) [reply client/rpc/src/state/mod.rs]: Done

**Review by cheme — APPROVED** [review]
LGTM

**Comment by bkchr** [comment]
bot merge

File: client/rpc-api/src/child_state/mod.rs [hunk 3]
```diff
@@ -68,4 +69,13 @@ pub trait ChildStateApi<Hash> {
 		key: StorageKey,
 		hash: Option<Hash>
 	) -> FutureResult<Option<u64>>;
+
+	/// Returns proof of storage for child key entries at a specific block's state.
+	#[rpc(name = "state_getChildReadProof")]
```
Comment (dvdplm) [inline client/rpc-api/src/child_state/mod.rs:74]: @mustermeiszer is there a specific reason this RPC is named `state_` and not `childstate_` like the other RPCs in this file?

Comment (mustermeiszer) [reply client/rpc-api/src/child_state/mod.rs:74]: TBH, it's a copy paste mistake from my side, as I grapped the logic from the 'state' impl. So renaming this to 'childstate' would probably be favorable.

Comment (dvdplm) [reply client/rpc-api/src/child_state/mod.rs:74]: Ok, cool. I've changed and added an alias to `state_getChildReadProof`. :)

**Pull request merged**